content/articles/fcm-with-spring-and-angular/index.md

```diff
@@ -0,0 +1,500 @@
+});
+```
+Finally, display these messages in our `app.component.html` file with a template like this:
+
+```angular2html
+<h1>Hello World</h1>
+<h3>These are your messages:</h3>
+<ul>
+ <li *ngFor="let message of messages">
+   <h3>{{message.title}}</h3>
+   <p>
+     {{message.body}}
+   </p>
+   <img [src]="message.iconUrl" alt="message-icon">
+ </li>
+</ul>
+```
+
+#### Receiving the notifications while the app is inactive
+
+As we mentioned in the project architecture, we need to use a service worker in the case of the app being closed. Firebase looks for a file from us called: `firebase-messaging-sw.js`. Firebase will take this file from us, and use it to generate the service worker. This file should go under the `src` folder and would have the following contents:
+
+```typescript
+importScripts('https://www.gstatic.com/Firebasejs/8.7.0/Firebase-app.js')
+importScripts('https://www.gstatic.com/Firebasejs/8.7.0/Firebase-messaging.js')
+
+// The object we pass as an argument is the same object we copied into the environment files
+Firebase.initializeApp({
+  apiKey: "...",
+  authDomain: "...",
+  projectId: "...",
+  storageBucket: "...",
+  messagingSenderId: "...",
+  appId: "...",
+  measurementId: "..."
+})
+
+const messaging = Firebase.messaging();
+```
+First, we need to import Firebase and Firebase messaging into the service worker file. You may be wondering why we use this weird ` importScripts` function to do this.  Service workers and other types of workers work differently than a normal javascript file. This is why they must use the `importScripts` function to import anything. Don’t worry about why this is the case, we don’t need to understand that. Anyways, with that installed, we call the `initializeApp` method passing the object we copied into the environment files. Then, all we need to do is create an object called *messaging* using the `messaging` method. This should handle all the magic of displaying notifications for us!
+
+With that created, try sending a post request to our server to send a notification while the app is inactive. In case you don’t know, you can do so using a tool like [Postman](https://www.postman.com/). You should see something like this on the corner of your monitor:
+
+![a sample notification](notification.png)
+
+### Conclusion
+
+In this guide, we went through the process of sending notifications in a full-stack Spring Boot and Angular project. In the backend, we set up a REST API to tell Firebase what notifications to send and where. In the frontend, we learned how to subscribe to receive notifications even when the app is closed. To best use this guide, try making your own full-stack project that uses these concepts. If you need help or a reference to what we did here, I created a [Github repository](https://github.com/john-amiscaray/Spring-Angular-Firebase-Cloud-Messaging) with all the code we wrote. Happy coding!
```

Comment:
If possible, add a `Further reading` section containing other references that might help the reader.